lib/pure/memfiles.nim

```diff
@@ -39,8 +46,19 @@ type
 
 {.deprecated: [TMemFile: MemFile].}
 
+proc checkConstructed(m: var MemFile) =
+  if (not m.initialized):
```

Comment:
Because default value of int is `0`, not INVALID_HANDLE. If `not initialized`, check `handle` is wrong, need initialize handle to INVALID_HANDLE before.

If we can constraint the type must initialize, or Nim has constructor, no need this.

In #3316, you suggest `wasOpen` flag, I think `initialized` more general for without constructor.